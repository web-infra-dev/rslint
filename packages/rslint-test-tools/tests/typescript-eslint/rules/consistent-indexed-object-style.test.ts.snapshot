exports[`consistent-indexed-object-style > invalid 1`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 10`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 12
        },
        "end": {
          "line": 1,
          "column": 43
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 11`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 20
        },
        "end": {
          "line": 1,
          "column": 42
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 12`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 20
        },
        "end": {
          "line": 1,
          "column": 51
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 13`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 19
        },
        "end": {
          "line": 1,
          "column": 41
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 14`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 17
        },
        "end": {
          "line": 1,
          "column": 39
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 15`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 19
        },
        "end": {
          "line": 1,
          "column": 50
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 16`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 17
        },
        "end": {
          "line": 1,
          "column": 48
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 17`] = `
{
  "diagnostics": [],
  "errorCount": 0,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 2`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 3`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 4`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 5`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 6`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 7`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 8`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 2,
          "column": 1
        },
        "end": {
          "line": 4,
          "column": 2
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;

exports[`consistent-indexed-object-style > invalid 9`] = `
{
  "diagnostics": [
    {
      "ruleName": "consistent-indexed-object-style",
      "messageId": "preferRecord",
      "message": "A record is preferred over an index signature.",
      "filePath": "src/virtual.ts",
      "range": {
        "start": {
          "line": 1,
          "column": 12
        },
        "end": {
          "line": 1,
          "column": 34
        }
      }
    }
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1
}
`;
