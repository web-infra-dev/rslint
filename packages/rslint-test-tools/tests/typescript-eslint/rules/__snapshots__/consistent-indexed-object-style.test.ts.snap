// Rstest Snapshot v1

exports[`consistent-indexed-object-style > invalid 1`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 2`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 3`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 4`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 5`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 6`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 7`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 8`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 9`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 34,
          "line": 1,
        },
        "start": {
          "column": 12,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 10`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 43,
          "line": 1,
        },
        "start": {
          "column": 12,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 11`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 42,
          "line": 1,
        },
        "start": {
          "column": 20,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 12`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 51,
          "line": 1,
        },
        "start": {
          "column": 20,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 13`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 41,
          "line": 1,
        },
        "start": {
          "column": 19,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 14`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 39,
          "line": 1,
        },
        "start": {
          "column": 17,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 15`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 50,
          "line": 1,
        },
        "start": {
          "column": 19,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 16`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 48,
          "line": 1,
        },
        "start": {
          "column": 17,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 17`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "An index signature is preferred over a record.",
      "messageId": "preferIndexSignature",
      "range": {
        "end": {
          "column": 31,
          "line": 1,
        },
        "start": {
          "column": 12,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 18`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "An index signature is preferred over a record.",
      "messageId": "preferIndexSignature",
      "range": {
        "end": {
          "column": 32,
          "line": 1,
        },
        "start": {
          "column": 15,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 19`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 34,
          "line": 1,
        },
        "start": {
          "column": 12,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 20`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 41,
          "line": 1,
        },
        "start": {
          "column": 12,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 21`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 51,
          "line": 1,
        },
        "start": {
          "column": 29,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 22`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 37,
          "line": 1,
        },
        "start": {
          "column": 12,
          "line": 1,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 23`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 24`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 2,
          "line": 4,
        },
        "start": {
          "column": 1,
          "line": 2,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 25`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "A record is preferred over an index signature.",
      "messageId": "preferRecord",
      "range": {
        "end": {
          "column": 38,
          "line": 3,
        },
        "start": {
          "column": 16,
          "line": 3,
        },
      },
      "ruleName": "consistent-indexed-object-style",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`consistent-indexed-object-style > invalid 26`] = `
{
  "diagnostics": [],
  "errorCount": 0,
  "fileCount": 1,
  "ruleCount": 1,
}
`;
